deltaspike/src/main/java/io/cucumber/deltaspike/DeltaSpikeObjectFactory.java

```diff
@@ -0,0 +1,47 @@
+package io.cucumber.deltaspike;
+
+import org.apache.deltaspike.cdise.api.CdiContainer;
+import org.apache.deltaspike.cdise.api.CdiContainerLoader;
+import io.cucumber.core.backend.ObjectFactory;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import java.util.Set;
+
+public class DeltaSpikeObjectFactory implements ObjectFactory {
```

Comment:
Note to self: Make final, add @API annotation

Reply:
can you me explain why you need this annotation here? or why you set it to stable and not to internal (other objectfactories)... for me this is not a api is it a implementation and nobody should rely on it?
btw. have done it, push will follow.

Reply:
There are a few use cases where decorating the object factory makes sense. Aslak wrote about one here:

https://cucumber.io/blog/polymorphic-step-definitions/

Unfortunately he used inheritance rather then composition. I want to allow for composition and forbid inheritance. 

The final prevents inheritance. Marking the class as part of the API renders it safe for composition.

Reply:
Ideally we'd have a different system for this.

Something that would be composeable with object factories with out being an object factory. But that is a design challenge that still needs to be solved.

A use case for this would be `PageFactory`. 


https://www.toolsqa.com/selenium-webdriver/page-object-pattern-model-page-factory/

Reply:
Thank for the explanation, with cdi some extensions should be possible with 'javax.enterprise.inject.spi.Extension' (use it on a closed project to integrate arquillian-drone)

Reply:
Aah. That also works.